### PR TITLE
[ Rudderstack + Growthbook ] tracking + A/B testing features

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

-   A/B test setup with Rudderstack

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/common/constants.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 const MAP_API_KEY = process.env.GATSBY_MAP_API_KEY
+const GATSBY_GROWTHBOOK_CLIENT_KEY =
+    process.env.GATSBY_GROWTHBOOK_CLIENT_KEY || 'sdk-0aHHaxKhwn0bP1HE'
```
Comment (habib-deriv) [inline src/common/constants.ts:3]:
```suggestion
    process.env.GATSBY_GROWTHBOOK_CLIENT_KEY'
```

File: src/features/hooks/analytic/use-analytic-events/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const { form_source, form_name } = useAnalyticData()
+
+    const analytic_events = {
+        [event]: useCallback((action: AnalyticsEventsTypes) => {
+            window.rudderanalytics.track(
+                event,
+                { action, form_source, form_name },
+                // for demo
+                () => {
+                    console.log(`${isMobile ? `${action} on mobile` : action}`, `${event} happened`)
```
Comment (habib-deriv) [inline src/features/hooks/analytic/use-analytic-events/index.ts:26]: Do we need the console?
```suggestion
                    console.log(`${isMobile ? `${action} on mobile` : action}`, `${event} happened`)
```

File: src/common/constants.ts (hunk 1 above)
Comment (NikitK-deriv) [reply src/common/constants.ts]: done, also add in vercel to build successfully

File: src/features/hooks/analytic/use-analytic-events/index.ts (hunk 2 above)
Comment (NikitK-deriv) [reply src/features/hooks/analytic/use-analytic-events/index.ts:25]: need for testing purpose

**Review by habib-deriv — APPROVED** [review]

**Pull request merged**